File: network/src/main/java/com/architecture/apilib/interceptor/CommonRequestInterceptor.java
package com.architecture.apilib.interceptor;

import androidx.annotation.NonNull;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * 通用的请求拦截器，
 */
public class CommonRequestInterceptor implements Interceptor {

    @NonNull
    @Override
    public Response intercept(Chain chain) throws IOException {
        Request.Builder builder = chain.request().newBuilder();
        builder.addHeader("Connection", "Keep-Alive");
        return chain.proceed(builder.build());
    }

}

File: network/src/main/java/com/architecture/apilib/interceptor/CommonRequestInterceptor.kt
package com.architecture.apilib.interceptor

import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

/**
 * 通用的请求拦截器，
 */
class CommonRequestInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val builder = chain.request().newBuilder()
        builder.addHeader("Connection", "Keep-Alive")
        return chain.proceed(builder.build())
    }
}
